### PR TITLE
add flag to manage $KUBCONFIG files

### PR DESCRIPTION
**Pull request by deads2k**
This adds a flag to `kubectl config *` commands to directly view and modify the .kubeconfig file referenced by $KUBECONFIG.

### CODE REVIEW

**Comment by jlowdermilk** [comment]
+1, I've been wanting this change. I actually think it should be default behavior; the envvar flag is unnecessary. If I set $KUBECONFIG, I expect kubectl to use it for both read and write.

**Comment by deads2k** [comment]
> If I set $KUBECONFIG, I expect kubectl to use it for both read and write.

`git config` operates against local config by default.  Admittedly, there isn't a environment variable file for git, but I think people might expect congruence.  It's a very easy change, but I do want to make sure it's an explicit one.

@smarterclayton Any thoughts?

**Comment by jlowdermilk** [comment]
I think write priority should match load order. The [loading rules](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/kubeconfig-file.md#loading-and-merging-rules) will prioritize a kubeconfig file from envvar over the local one. This means that with current behavior, if I set $KUBECONFIG, then run `kubectl set-context foo`, that context is potentially ignored by subsequent calls because I set it in a local `.kubeconfig`, but I'm reading from the envvar `.kubeconfig` first.

**Comment by deads2k** [comment]
I'm convinced.  `--local` is now false by default.  `--envvar` still exists as an explicit option to clearly specify intent.  Default behavior is 
 1.  $KUBECONFIG if it is set
 1.  ./.kubeconfig if it is not.

**Comment by jlowdermilk** [comment]
I'm fine with keeping `--envvar` as an explicit option, as long as we fail a command if `--envvar` was specified, but the variable is unset.

Also, this is a bit tangent, but it doesn't look like we prevent the user from specifying multiple flags for kubeconfig source. i.e. we should error out on a command like
```
kubectl config set-context foo --global --local
```
Am I just missing that error code?

File: pkg/kubectl/cmd/config/config.go [hunk 1]
```diff
@@ -66,17 +68,32 @@ func (o *pathOptions) getStartingConfig() (*clientcmdapi.Config, string, error)
...
+
 	case o.global:
 		filename = os.Getenv("HOME") + "/.kube/.kubeconfig"
 		config = getConfigFromFileOrDie(filename)
 
-	case len(o.specifiedFile) > 0:
-		filename = o.specifiedFile
+	case o.envvar:
+		filename = os.Getenv(clientcmd.RecommendedConfigPathEnvVar)
 		config = getConfigFromFileOrDie(filename)
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/config/config.go]: log fatal if len(filename) == 0

**Comment by deads2k** [comment]
> I'm fine with keeping --envvar as an explicit option, as long as we fail a command if --envvar was specified, but the variable is unset.

Done.



> it doesn't look like we prevent the user from specifying multiple flags for kubeconfig source

Nope, looks like an oversight.  I added it.

**Comment by jlowdermilk** [comment]
lgtm.

**Pull request merged**